src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotpyeIterator.java

```diff
@@ -0,0 +1,108 @@
+package org.broadinstitute.hellbender.tools.walkers.genotyper;
+
+import org.broadinstitute.hellbender.tools.walkers.genotyper.GLVectorSizeCache;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+/**
+ * Handles addressing general ploidy genotype likelihoods. We store likelihoods as a map
+ * of form int[] -> double (to be more precise, IntArrayWrapper -> Double).
+ * For a given ploidy (chromosome count) and number of alleles, we need a form to iterate deterministically
+ * across all possible allele conformations, that is, all possible ways in which N whole numbers will sum to P,
+ * where N is number of alleles and P is number of chromosomes.
+ * For example, with P=2,N=2, ploidy = 2 iterator will produce
+ * alleleCounts = [2 0 ] [1 1] [ 0 2]
+ *
+ *
+ */
+public final class GenotpyeIterator {
```

Comment:
@vruano I agree with you that this functionality is redundant with that of `GenotypeAlleleCounts` (as invoked by `GenotypeLikelihoodCalculator`.  This `GenotypeIterator` is simply a port of the former nested `SumIterator` class inside `GeneralPloidyExactAFCalculator`.  I would be quite happy to take the next step and eliminate one of the implementations.

However, it's not clear to me which one to get rid of.  The code in `GenotypeIterator` is more concise and its representation -- `{count of allele 0, count of allele 1, count of allele 2 . . .}` -- is more direct than the representation `{index of first allele to appear in this genotype, count of this allele, index of second allele, . . .}`  Then again, in the case of a large number of alleles this sparse representation would improve performance.  And `GenotypeAlleleCounts` has some additional functionality that may be important.

Since you are much more familiar with this code, I'm going to trust your opinion.  Which one should we keep?